src/main/java/uk/gov/hmcts/befta/util/MapVerifier.java

```diff
@@ -186,12 +184,16 @@ private void collectBadValueMessagesFromCollection(String fieldPrefix, String fi
         }
     }
 
-    private Object compareValues(String fieldPrefix, String commonKey, Object expectedValue,
+    private Object compareValues(String commonKey, Object expectedValue,
             Object actualValue, int currentDepth, int maxMessageDepth) {
         boolean justCompare = currentDepth > maxMessageDepth;
-        if (expectedValue instanceof String && ANY.equalsIgnoreCase((String) expectedValue)) {
+        if (expectedValue instanceof String &&
+                ANYTHING_IF_EXISTS.getValue().equalsIgnoreCase((String) expectedValue)) {
             return Boolean.TRUE;
-        } else if (expectedValue == actualValue) {
+        } else if(expectedValue instanceof String && isValidFormat((String) expectedValue, actualValue)) {
+            return Boolean.TRUE;
```

Comment:
these 2 checks could be combined in one method like
private boolean actualAcceptedByPlaceholder(String strExpectedValue, Object actualValue) {

Reply:
Changed